src/NLog/Layouts/LayoutParser.cs

```diff
@@ -438,7 +441,7 @@ private static LayoutRenderer GetLayoutRenderer(ConfigurationItemFactory configu
             {
                 if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions ?? LogManager.ThrowExceptions)
                 {
-                    throw;  // TODO NLog 5.0 throw NLogConfigurationException. Maybe also include entire input layout-string (if not too long)
+                    throw new NLogConfigurationException("Error parsing layout {0}", name);
```

Comment:
Instead of having the same log formatting twice, then one could consider this:

```c#
var configException = new NLogConfigurationException(ex, $"Error parsing layout {name}");
if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions ?? LogManager.ThrowExceptions)
     throw configException;

InternalLogger.Warn(ex, configException.Message);
```

Reply:
Is it really a rethrow in this case? I thought it should only be used like this:

https://github.com/NLog/NLog/blob/dev/src/NLog/Internal/AssemblyHelpers.cs#L176-L180

Reply:
It can also be used like this: https://github.com/NLog/NLog/blob/6198b8696c1bf571e2266c599063557fe486dcc6/src/NLog/Config/LoggingConfigurationElementExtensions.cs#L108